### PR TITLE
Verify existence of doc.go file

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

File: sdk/internal/atomic/doc.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package atomic
```
Comment (RickWinter) [inline sdk/internal/atomic/doc.go:7]:
```suggestion
package atomic

```

File: eng/pipelines/templates/steps/analyze.yml [hunk 2]
```diff
@@ -24,6 +24,16 @@ steps:
     failOnStderr: false
     workingDirectory: '${{parameters.GoWorkspace}}'
 
+  - pwsh: |
+      $modDirs = (./eng/scripts/get_module_dirs.ps1 -serviceDir $(SCOPE))
+      foreach ($md in $modDirs) {
+        pushd $md
+        Get-ChildItem doc.go
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
+      }
```
Comment (benbp) [inline eng/pipelines/templates/steps/analyze.yml:33]: `$LASTEXITCODE` only applies to exit codes from external process calls. Powershell commands/functions are handled via `throw`/exceptions. You could also use a `Test-Path` conditional here.

```suggestion
      foreach ($md in $modDirs) {
        Get-ChildItem $md/doc.go
      }

File: eng/pipelines/templates/steps/analyze.yml [hunk 3]
```diff
@@ -24,6 +24,16 @@ steps:
     failOnStderr: false
     workingDirectory: '${{parameters.GoWorkspace}}'
 
+  - pwsh: |
+      $modDirs = (./eng/scripts/get_module_dirs.ps1 -serviceDir $(SCOPE))
```
Comment (benbp) [inline eng/pipelines/templates/steps/analyze.yml:28]:
```suggestion
      $modDirs = ./eng/scripts/get_module_dirs.ps1 -serviceDir $(SCOPE)
```

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/steps/analyze.yml [hunk 4]
```diff
@@ -24,6 +24,14 @@ steps:
     failOnStderr: false
     workingDirectory: '${{parameters.GoWorkspace}}'
 
+  - pwsh: |
+      $modDirs = ./eng/scripts/get_module_dirs.ps1 -serviceDir $(SCOPE)
+      foreach ($md in $modDirs) {
+        Get-ChildItem $md/doc.go
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:30]: Is the idea that you want this to fail if it isn't present? If so perhaps we should do a Test-Path and give a nicer error message if it is missing.

**Pull request merged**